### DISCUSSION TITLE
DBT override profiles dir option

### DISCUSSION THREAD

**Discussion by apantaz** [AltimateAI/vscode-dbt-power-user#953] [Q&A] [answered]
Hello everyone. Previously, there was an option to override the profiles directory path, but it seems that option has been removed, as indicated by https://github.com/AltimateAI/vscode-dbt-power-user/pull/880. Finding an alternative is crucial for my dbt project, as I need to ensure that the profiles.yml for my local .dbt folder takes precedence over the one for Github actions to avoid conflicts. Any suggestions for achieving this are appreciated.

**Update**: Resolved. The problem was that I set the env variables into the .bashrc file. Needed to add them in the VS Code settings.json file.

**Comment by anandgupta42** [comment, marked as answer]
The setting is now available as an environment variable. This was done to be consistent with how dbt does things.

It has been documented in the docs : https://docs.myaltimate.com/setup/optConfig/#default-dbt-env-support

**Reply by pSpitzner** [reply]
Hi everyone,

I would like to chime in that the env variables are not a complete replacement for this.
In particular, relative paths are not respected, see this PR: #1573

Thus, it is currently not possible to set the profiles dir relative to the workspace folder.
Say, for example, I want to have the profiles.yml in a config folder: `./config/profiles` where the config folder sits next to the main dbt project folder.

A (rather involved) workaround is to have a .env file.
Importantly, _even then the path has to be set absolutely_, making this less portable.

File Structure:
```
./config/
    myproject.code-workspace
    profiles.yml
    .env
./project_with_models_etc/
    models
    ...
 
```

`${workspaceFolder}/config/.env`

```bash
workspaceFolder=/Users/paul/set/path/absolute

# using ./config will not work, see PR 1573
DBT_PROFILES_DIR=${workspaceFolder}/config/
DBT_PROJECT_DIR=${workspaceFolder}/project_with_models_etc/
```

Then i need to source the .env in the vs code workspace file

`myproject.code-workspace`
```json
{
    "folders": [
        {
            "path": ".."
        }
    ],
    "settings": {
        "python.envFile": "${workspaceFolder}/config/.env",
     }
}

```
